cpp/src/parquet/metadata.cc

```diff
@@ -87,20 +87,28 @@ std::string ParquetVersionToString(ParquetVersion::type ver) {
 template <typename DType>
 static std::shared_ptr<Statistics> MakeTypedColumnStats(
     const format::ColumnMetaData& metadata, const ColumnDescriptor* descr) {
+  DCHECK(metadata.__isset.statistics);
+  int64_t num_values = 0;
```

Comment:
I've handle it here. Maybe it's not right. But I think it's compatible with previous code